core/src/dispatch/mod.rs

```diff
@@ -202,6 +209,58 @@ impl Default for NetworkConfig {
     }
 }
 
+/// A port providing `NetworkStatusUpdates´ to listeners.
+pub struct NetworkStatusPort;
+impl Port for NetworkStatusPort {
+    type Indication = NetworkStatus;
+    type Request = NetworkStatusRequest;
+}
+
+/// Information regarding changes to the systems connections to remote systems
+#[derive(Clone, Debug)]
+pub enum NetworkStatus {
+    /// Indicates that a connection has been established to the remote system
+    ConnectionEstablished(SystemPath),
+    /// Indicates that a connection has been lost to the remote system.
+    /// The system will automatically try to recover the connection for a configurable amount of
+    /// retries. The end of the automatic retries is signalled by a `ConnectionDropped` message.
+    ConnectionLost(SystemPath),
+    /// Indicates that a connection has been dropped and no more automatic retries to re-establish
+    /// the connection will be attempted and all queued messages have been dropped.
+    ConnectionDropped(SystemPath),
+    /// Indicates that a connection has been gracefully closed.
+    ConnectionClosed(SystemPath),
+    /// The list includes all remote systems which the system is currently connected to.
+    /// Sent as response to `NetworkStatusRequest::RemoteSystems`.
+    ConnectedSystems(Vec<SystemPath>),
```

Comment:
Oooh, don't do this! Remember that indications will be cloned when multiple ports are connected.
Normally you should put this behind an Arc, i.e. `Arc<[SystemPath]>`, but since this is actually a response to a request, I think it should be using the Ask mechanism, tbh. We don't want one component to ask for the connected systems and everyone to get spammed with replies they didn't ask for.